verilog/analysis/checkers/always_ff_non_blocking_rule.cc

```diff
@@ -97,31 +100,71 @@ void AlwaysFFNonBlockingRule::HandleSymbol(const verible::Symbol& symbol,
   static const Matcher asgn_incdec_matcher{NodekIncrementDecrementExpression()};
   static const Matcher ident_matcher{NodekUnqualifiedId()};
 
+  std::vector<AutoFix> autofixes;
+
   // Rule may be waived if complete lhs consists of local variables
   //  -> determine root of lhs
-  const verible::Symbol* check_root = nullptr;
+  const verible::Symbol *check_root = nullptr;
 
   verible::matcher::BoundSymbolManager symbol_man;
   if (asgn_blocking_matcher.Matches(symbol, &symbol_man)) {
-    if (const auto* const node =
-            verible::down_cast<const verible::SyntaxTreeNode*>(&symbol)) {
-      check_root =
-          /* lhs */ verible::down_cast<const verible::SyntaxTreeNode*>(
-              node->children()[0].get());
-    }
+    const verible::SyntaxTreeNode *node = &verible::SymbolCastToNode(symbol);
+    check_root = verilog::GetNetVariableAssignmentLhs(*node);
+
+    const verible::SyntaxTreeLeaf *leaf =
+        verible::GetSubtreeAsLeaf(symbol, NodeEnum::kNetVariableAssignment, 1);
+
+    autofixes.emplace_back(AutoFix(
+        "Substitute blocking assignment '=' for nonblocking assignment '<='",
+        {leaf->get(), "<="}));
   } else {
     // Not interested in any other blocking assignments unless flagged
     if (!catch_modifying_assignments_) return;
 
+    // These autofixes require substituting the whole expression
+    absl::string_view original = verible::StringSpanOfSymbol(symbol);
     if (asgn_modify_matcher.Matches(symbol, &symbol_man)) {
-      if (const auto* const node =
-              verible::down_cast<const verible::SyntaxTreeNode*>(&symbol)) {
-        check_root =
-            /* lhs */ verible::down_cast<const verible::SyntaxTreeNode*>(
-                node->children()[0].get());
-      }
+      const verible::SyntaxTreeNode *node = &verible::SymbolCastToNode(symbol);
+      const verible::SyntaxTreeNode &lhs = *GetAssignModifyLhs(*node);
+      const verible::SyntaxTreeNode &rhs = *GetAssignModifyRhs(*node);
+      const verible::SyntaxTreeLeaf &operator_ =
+          *GetAssignModifyOperator(*node);
+
+      check_root = &lhs;
+
+      const std::string fix =
+          absl::StrCat(verible::StringSpanOfSymbol(lhs),
+                       " <= ", verible::StringSpanOfSymbol(lhs), " ",
+                       verible::StringSpanOfSymbol(operator_).substr(0, 1), " ",
+                       verible::StringSpanOfSymbol(rhs), ";");
+
+      autofixes.emplace_back(
+          AutoFix("Substitute assignment operator for equivalent "
+                  "nonblocking assignment",
+                  {original, fix}));
     } else if (asgn_incdec_matcher.Matches(symbol, &symbol_man)) {
-      check_root = &symbol;
+      const verible::SyntaxTreeNode *operand =
+          GetIncrementDecrementOperand(symbol);
+      const verible::SyntaxTreeLeaf *operator_ =
```

Comment:
why underscore in the `operator_` name ?

Reply:
```operator``` is a reserved keyword, so I couldn't use it.